src/private.rs

```diff
@@ -52,14 +41,14 @@ impl RedactFlags {
     /// Maximum number of characters to expose at the beginning and end of a partial redact.
     const MAX_PARTIAL_EXPOSE: usize = 3;
 
-    fn redact_partial(&self, str: &str, redacted: &mut String) {
+    pub(crate) fn redact_partial(&self, fmt: &mut std::fmt::Formatter, str: &str) -> std::fmt::Result {
         let count = str.chars().filter(|char| char.is_alphanumeric()).count();
         if count < Self::MIN_PARTIAL_CHARS {
             for char in str.chars() {
                 if char.is_alphanumeric() {
-                    redacted.push(self.redact_char);
+                    fmt.write_char(self.redact_char)?;
```

Comment:
I'm a bit worried about the performance of using many write_chars instead of batching the writes together. No metrics on this though

Reply:
Yeah I was also concerned about that actually. One would hope that there would be a buffered writer in the middle somewhere. Will need to investigate

Reply:
I guess that depends on the formatter impl. I'm not sure if we should be doing our own buffering, if a formatter doesn't do buffering it's not our problem. On the other hand we could maybe preallocate memory more efficiently? Idk